tests/templates/test_subroutines/test_controlled_sequence.py

```diff
@@ -0,0 +1,408 @@
+        op = qml.ControlledSequence(qml.PauliX(0), control=[1])
+        assert op.has_matrix is False
+        assert op.base.has_matrix is True
+
+
+class TestMethods:
+    def test_repr(self):
+        """Test that the operator repr is as expected"""
+        op = qml.ControlledSequence(qml.RX(0.25, wires=3), control=[0, 1, 2])
+        assert repr(op) == f"ControlledSequence({op.base}, control=[0, 1, 2])"
+
+    def test_map_wires(self):
+        """Test mapping wires works as expected, including mapping wires on the base operator"""
+
+        op = qml.ControlledSequence(qml.CNOT([0, 1]), control=[2, 3])
+        wire_map = {0: "a", 1: "b", 2: "c", 3: "d"}
+
+        new_op = op.map_wires(wire_map)
+
+        assert type(new_op) == type(op)
+        assert type(new_op.base) == type(op.base)
+        assert new_op.data == op.data
+
+        assert new_op.wires == Wires(["a", "b", "c", "d"])
+        assert new_op.base.wires == Wires(["a", "b"])
+        assert new_op.control == Wires(["c", "d"])
+
+    def test_compute_decomposition(self):
+        """Test that the decomposition is as expected"""
+        base = qml.RZ(4.3, 1)
+        control_wires = [0, 2, 3]
+
+        decomp = qml.ControlledSequence.compute_decomposition(base, control_wires=control_wires)
+
+        assert len(decomp) == len(control_wires)
+        for i, op in enumerate(decomp):
+            assert qml.equal(op.base.base, base)
+            assert isinstance(op, qml.ops.Pow)
+            assert op.z == 2 ** (len(control_wires) - i - 1)
+
+        for op, w in zip(decomp, control_wires):
+            assert op.base.control_wires == Wires(w)
+
+    def test_decomposition(self):
+        op = qml.ControlledSequence(qml.RX(0.25, wires=3), control=["a", 1, "blue"])
+
+        decomp = op.decomposition()
+        expected_decomp = qml.ControlledSequence.compute_decomposition(
+            qml.RX(0.25, wires=3), ["a", 1, "blue"]
+        )
+
+        for op1, op2 in zip(decomp, expected_decomp):
+            assert op1 == op2
+
+
+class TestIntegration:
+    """Tests that the ControlledSequence is executable and differentiable in a QNode context"""
+
+    @staticmethod
+    def circuit(x):
+        """Test circuit"""
+        qml.PauliX(2)
+        qml.ControlledSequence(qml.RX(x, wires=3), control=[0, 1, 2])
+        return qml.probs(wires=range(4))
+
+    x = np.array(0.25)
+    # not calculated analytically, we are only ensuring that the results are consistent accross interfaces
+    exp_result = np.array([0, 0, 0.9835, 0.0165, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+    exp_jac = np.array([0, 0, -0.12342829, 0.12342829, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+
+    def test_qnode_numpy(self):
+        """Test that the QNode executes with Numpy."""
+        dev = qml.device("default.qubit")
+        qnode = qml.QNode(self.circuit, dev, interface=None)
+
+        res = qnode(self.x)
+        assert res.shape == (16,)
+        assert np.allclose(res, self.exp_result, atol=0.002)
```

Comment:
Why this large atol?

Reply:
Otherwise the test fails - this test is relatively inaccurate because it also has an implementation using shots rather than analytic estimation, which is probabilistic, and the output varies. 

The results are on the order of 0.1, so plus-minus 0.002 is not so bad anyway.